tensorflow_lite_support/c/task/processor/base_options.h

```diff
@@ -0,0 +1,77 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_LITE_SUPPORT_C_TASK_PROCESSOR_BASE_OPTIONS_H_
+#define TENSORFLOW_LITE_SUPPORT_C_TASK_PROCESSOR_BASE_OPTIONS_H_
+
+#include <stdint.h>
+
+// Defines C Structs for Base Options Shared by all tasks.
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+// Holds settings to configure cpu execution.
+typedef struct TfLiteCpuSettings {
```

Comment:
Thanks for implementing these structs! 

Just a reminder, please don't add more features in this PR. It is already very large. In general, small PRs (less than 5 files and less than 300 lines of code) are preferred. It helps reduce complexity in both development and code review. In the future, we can try adding code in the following order.
* APIs: methods declarations (.h files), protos, BUILD targets.
* Implementations: .cc files and some simple smoke tests.
* More thorough tests.